pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/actions/ContextMenuAction.kt

```diff
@@ -5,33 +5,74 @@ package com.amos.pitmutationmate.pitmutationmate.actions
 
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.actionSystem.CommonDataKeys
-import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiClass
 import com.intellij.psi.PsiClassOwner
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.psi.KtClass
 
 class ContextMenuAction : RunConfigurationAction() {
     override fun actionPerformed(e: AnActionEvent) {
+        val editor = e.getData(CommonDataKeys.EDITOR)
         val psiFile = e.getData(CommonDataKeys.PSI_FILE)
-        println("ContextMenuAction: actionPerformed for file $psiFile")
-        val psiClasses = (psiFile as PsiClassOwner).classes
-        val fqns = mutableListOf<String>()
-        for (psiClass in psiClasses) {
-            val fqn = psiClass.qualifiedName
-            if (fqn != null) {
-                fqns.add(fqn)
-                println("ContextMenuAction: detected class '$fqn'")
+        if (e.place == "EditorPopup") {
+            println("ContextMenuAction: actionPerformed in EditorPopup for file $psiFile")
+            val psiElement = psiFile?.findElementAt(editor?.caretModel!!.offset)
+            val selectedClass = findEnclosingClass(psiElement)
+            if (selectedClass != null) {
+                var classFQN = ""
+                if (selectedClass is PsiClass) {
+                    classFQN = selectedClass.qualifiedName.toString()
+                }
+                if (selectedClass is KtClass) {
+                    classFQN = selectedClass.fqName.toString()
+                }
+
+                println("ContextMenuAction: selected class is $classFQN.")
+                updateAndExecuteRunConfig(classFQN, e.project!!, editor)
             }
         }
-        val editor = e.getData(CommonDataKeys.EDITOR)
-        updateAndExecuteRunConfig(fqns.first(), e.project!!, editor)
+        if (e.place == "ProjectViewPopup") {
+            println("ContextMenuAction: actionPerformed in ProjectViewPopup for file $psiFile")
+            val psiClasses = (psiFile as PsiClassOwner).classes
+            var classFQNs: String = ""
```

Comment:
wouldn't it be better to use an array here and consequently pass it later on to the executor?

Reply:
It seemed easier to use strings since it would later on be converted to string anyway, but if you think it's better to use an array I can change it.

Reply:
Well it doesn't really make a difference. I think it's cleaner to do the decision of how the classpaths are made into a string in one place and only at the end of the processing loop but there is no specific need for that.
If thats too much work keep it as is.

It would only allow the override notificaiton to provide a better notification string (applied target class xyz vs applied target classes xyz and abc)

Reply:
Actually it doesn't seem so easy to change. The variable classFQNOption in RunConfigurationOptions.kt cannot so easily be converted to array of strings. Or at least I would need time to look into how this can be done, which I won't have before the end of this sprint. This can maybe be a small refactoring ticket @emuguy1 @Felix-012 ?